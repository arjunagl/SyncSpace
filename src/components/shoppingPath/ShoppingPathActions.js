export const completeShoppingPathAction = () => ({
    type: 'COMPLETE_SHOPPING'
});

