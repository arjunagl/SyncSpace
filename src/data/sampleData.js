export const AppliedShoppingListsSampleData = [
    {
        ShoppingListId: '1',
        Item: {
            Id: '1',
            Name: 'Apples'
        },
        Location: {
            Name: 'Isle 1',
            Description: 'Next to Oranges'
        },
    }, {
        ShoppingListId: '1',
        Item: {
            Id: '2',
            Name: 'Pears'
        },
        Location: {
            Name: 'Isle 1',
            Description: 'Next to Oranges'
        },
    },
    {
        ShoppingListId: '1',
        Item: {
            Id: '3',
            Name: 'Oranges'
        },
        Location: {
            Name: 'Isle 2',
            Description: 'Next to Oranges'
        },
    },
    {
        ShoppingListId: '1',
        Item: {
            Id: '4',
            Name: 'Grapes'
        },
        Location: {
            Name: 'Isle 1',
            Description: 'Next to Oranges'
        },
    },
    {
        ShoppingListId: '1',
        Item: {
            Id: '5',
            Name: 'Carrots'
        },
        Location: {
            Name: 'Isle 4',
            Description: 'Next to Oranges'
        },
    },
    {
        ShoppingListId: '1',
        Item: {
            Id: '6',
            Name: 'Apricots'
        },
        Location: {
            Name: 'Isle 3',
            Description: 'Next to Oranges'
        },
    }
];
