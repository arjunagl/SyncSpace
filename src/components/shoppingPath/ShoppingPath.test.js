import React from 'react';
import { mount, shallow } from 'enzyme';
import { Provider } from 'react-redux';
import sinon from 'sinon';
import toJson from 'enzyme-to-json';
import configureStore from 'redux-mock-store';
import ShoppingPathComponentContainer, { ShoppingPathComponent } from './ShoppingPath';
import { AppliedShoppingListsSampleData } from '../../data/sampleData';
import { completeShoppingPathAction } from './ShoppingPathActions';

describe('<ShoppingPathComponent />', () => {
    let store;

    beforeEach(() => {
        const mockStore = configureStore();
        const getState = {
            syncSpaceReducer: {
                AppliedShoppingLists: AppliedShoppingListsSampleData
            }
        }; // initial state of the store         
        store = mockStore(getState);
    });

    it('renders the component properly', () => {
        const component = shallow(
            <ShoppingPathComponent
                store={store}
                AppliedShoppingLists={AppliedShoppingListsSampleData}
            />
        );
        expect(toJson(component)).toMatchSnapshot();
    });

    it('calls \'onCompleteShoppingClicked\' when the user clicks the button to complete shopping', () => {
        const onCompleteShoppingClickedSpy = sinon.spy();
        const onSaveShoppingClickedSpy = sinon.spy();

        const shppingPathWrapper =
            mount(
                <Provider store={store}>
                    <ShoppingPathComponent
                        AppliedShoppingLists={AppliedShoppingListsSampleData}
                        onCompleteShoppingClicked={onCompleteShoppingClickedSpy}
                        onSaveShoppingClicked={onSaveShoppingClickedSpy}
                    />
                </Provider>
            );

        const completeShoppingbutton = shppingPathWrapper.find('#completeShoppingbutton');
        completeShoppingbutton.simulate('click');
        expect(onCompleteShoppingClickedSpy.called).toEqual(true);
    });

    it('calls \'onSaveShoppingClicked\' when the user clicks the button to save shopping', () => {
        const onCompleteShoppingClickedSpy = sinon.spy();
        const onSaveShoppingClickedSpy = sinon.spy();

        const shppingPathWrapper =
            mount(
                <Provider store={store}>
                    <ShoppingPathComponent
                        AppliedShoppingLists={AppliedShoppingListsSampleData}
                        onCompleteShoppingClicked={onCompleteShoppingClickedSpy}
                        onSaveShoppingClicked={onSaveShoppingClickedSpy}
                    />
                </Provider>
            );

        const completeShoppingbutton = shppingPathWrapper.find('#saveShoppingbutton');
        completeShoppingbutton.simulate('click');
        expect(onSaveShoppingClickedSpy.called).toEqual(true);
    });

    it('dispatches the completeShoppingPath action when complete shopping is clicked', () => {
        const shoppingPathContainerWrapper =
            mount(
                <Provider store={store}>
                    <ShoppingPathComponentContainer />
                </Provider>
            );

        const completeShoppingbutton = shoppingPathContainerWrapper.find('#completeShoppingbutton');
        completeShoppingbutton.simulate('click');
        const actions = store.getActions();
        expect(actions[0]).toEqual(completeShoppingPathAction());
    });

    it('selects registers the shopping list item when the user checks the item', () => {
        const onCompleteShoppingClickedSpy = sinon.spy();
        const onSaveShoppingClickedSpy = sinon.spy();

        const shoppingPathWrapper =
            mount(
                <Provider store={store}>
                    <ShoppingPathComponent
                        AppliedShoppingLists={AppliedShoppingListsSampleData}
                        onCompleteShoppingClicked={onCompleteShoppingClickedSpy}
                        onSaveShoppingClicked={onSaveShoppingClickedSpy}
                    />
                </Provider>
            );

        //Select a shopping list item
        const shoppingItem = shoppingPathWrapper.find('input[type=\'checkbox\']#1');
        shoppingItem.simulate('click');
        console.log('abcd');
    });
});
