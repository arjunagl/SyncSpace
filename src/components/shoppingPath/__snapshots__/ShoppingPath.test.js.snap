// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ShoppingPathComponent /> renders the component properly 1`] = `
<div>
  <div>
    <div>
      Isle 1
      <div>
        <div>
          <input
            id="1"
            type="checkbox"
          />
          <label
            htmlFor="1"
          >
            Apples
            <span>
              Next to Oranges
            </span>
          </label>
        </div>
        <div>
          <input
            id="2"
            type="checkbox"
          />
          <label
            htmlFor="2"
          >
            Pears
            <span>
              Next to Oranges
            </span>
          </label>
        </div>
        <div>
          <input
            id="4"
            type="checkbox"
          />
          <label
            htmlFor="4"
          >
            Grapes
            <span>
              Next to Oranges
            </span>
          </label>
        </div>
      </div>
    </div>
  </div>
  <div>
    <div>
      Isle 2
      <div>
        <div>
          <input
            id="3"
            type="checkbox"
          />
          <label
            htmlFor="3"
          >
            Oranges
            <span>
              Next to Oranges
            </span>
          </label>
        </div>
      </div>
    </div>
  </div>
  <div>
    <div>
      Isle 3
      <div>
        <div>
          <input
            id="6"
            type="checkbox"
          />
          <label
            htmlFor="6"
          >
            Apricots
            <span>
              Next to Oranges
            </span>
          </label>
        </div>
      </div>
    </div>
  </div>
  <div>
    <div>
      Isle 4
      <div>
        <div>
          <input
            id="5"
            type="checkbox"
          />
          <label
            htmlFor="5"
          >
            Carrots
            <span>
              Next to Oranges
            </span>
          </label>
        </div>
      </div>
    </div>
  </div>
  <div>
    <Connect(ButtonComponent)
      content="Complete shopping"
      id="completeShoppingbutton"
      style={
        Object {
          "margin": "10px 10px 0px 0px",
        }
      }
    />
    <Connect(ButtonComponent)
      content="Save for later"
      id="saveShoppingbutton"
      style={
        Object {
          "margin": "10px 10px 0px 0px",
        }
      }
    />
  </div>
</div>
`;
